apps/server/src/modules/sharing/uc/share-token.uc.ts

```diff
@@ -125,10 +164,19 @@ export class ShareTokenUC {
 
 		const user = await this.authorizationService.getUserWithPermissions(userId);
 
-		this.authorizationService.checkAllPermissions(user, [Permission.COURSE_CREATE]);
+		let requiredPermissions: string[] = [];
+		switch (parentType) {
+			case ShareTokenParentType.Course:
+				requiredPermissions = [Permission.COURSE_CREATE];
+				break;
+			default:
+				requiredPermissions = [Permission.TOPIC_CREATE];
```

Comment:
"courseTeacher" and "courseSubstitutionTeacher" have "LESSONS_CREATE" and "TOPIC_CREATE" permissions.  "teacher" only has "TOPIC_CREATE" permission. If I put "LESSONS_CREATE" here, another teacher won't be able to import this course. I used "TOPIC_CREATE" (topic=lesson from what I understand) because it was used in copy lesson already (https://github.com/hpi-schul-cloud/schulcloud-server/blob/main/apps/server/src/modules/learnroom/uc/lesson-copy.uc.ts#L25).